x/stakeibc/keeper/lsm.go

```diff
@@ -160,24 +160,32 @@ func (k Keeper) GetValidatorFromLSMTokenDenom(denom string, validators []*types.
 
 // Checks if we need to issue an ICQ to check if a validator was slashed
 // The query runs at periodic intervals defined by the ValidatorSlashQueryInterval
-func (k Keeper) ShouldCheckIfValidatorWasSlashed(ctx sdk.Context, validator types.Validator, stakeAmount sdkmath.Int) bool {
+// The interval is represented as percent of TVL
+// (e.g. 1% means every LS that causes the progress to breach 1% of TVL triggers the query)
+func (k Keeper) ShouldCheckIfValidatorWasSlashed(
+	ctx sdk.Context,
+	validator types.Validator,
+	totalHostZoneStake sdkmath.Int,
+	transactionStakeAmount sdkmath.Int,
+) bool {
 	params := k.GetParams(ctx)
-	queryInterval := sdk.NewIntFromUint64(params.ValidatorSlashQueryInterval)
+	queryInterval := sdk.NewDecWithPrec(int64(params.ValidatorSlashQueryInterval), 2) // percentage
+	checkpoint := queryInterval.Mul(sdk.NewDecFromInt(totalHostZoneStake)).TruncateInt()
```

Comment:
One other issue, the totalHostZoneStake can go up but also can go down if people choose to unbond. So imagine oldProgress and newProgress which are both quantities that only increase from transactions adding stake would have been on different sides of the line like the example in the comment below, but a change in the totalHostZoneStake going down now makes them both on the same side...

ex:  previous totalHostZoneStake 100000 --> previous checkpoint 1000
oldprogress 900 new progress 1100
but before this staking transaction someone unbonded 10000 making 
current totalHostZoneStake 90000 --> current checkpoint 900
then oldprgress/checkpoint = 1 and newprogress/checkpoint = 1.222 => 1 on int truncate
So this example should have triggered but because of the unbond it would not!!!

Reply:
It's a good flag that the checkpoint can be reduced via unbondings.

Could you expand on this example? I rewrote my understanding of your example. As it's written below I think it's expected behavior. Basically unbondings can lower the checkpoint.

ex: I think in your example someone is liquid staking 200 (based on "oldprogress 900 new progress 1100")
- Progress 900
- totalHostZoneStake 100000 --> checkpoint 1000
- Unbonding 10000
- totalHostZoneStake 90000 --> checkpoint 900
- LSM liquid stake 200 -> Progress 1100 ->  newprogress/checkpoint = 1.22 -> Slash Query issued

Reply:
My point is that Slash Query would not get issued.  It is only issued when the old/checkpoint is different from new/checkpoint but because of the change in checkpoint, now both of these are 1

Reply:
The way to fix these issues is to calculate once and store the value for checkpoint at the time of the last slash query instead of computing the checkpoint dynamically each time we do the check.

Reply:
yeah this is a good point, lemme think on it for a bit

Reply:
So while I do think it's possible to miss a query because of this, I don't think this is a huge concern considering the following: 
  1.  The TVL only has the potential to decrease when we submit the unbonding ICAs which is only once per 3-5 days
  2. The unbonding amount would have to decrease more than the amount of new stake we've had for that epoch (which is 100% possible and likely will happen at some point, but in practice has not happened yet)
  3. The worst case impact of this is that the query interval is twice as wide as it should be, but the 1% threshold is already so conservative so a bump to 2% is not concerning
  
So considering the compounding unlikeliness of this happening, with the small margin of impact, I think it's fine to safely ignore this. But curious to hear your guy's thoughts.

Reply:
I disagree but maybe I don't have the full info.  I think a malicious attacker could indefinitely keep us from correctly checking the slash query and the 3-5 day window makes it more complicated but not impossible.  The point of the example I was showing is that when the amount is just about to overflow the checkpoint, unbonding any can put both onto the over 1.0 ratio case.  But there is no reason they can't do this again and again.  Also keep in mind that these accumulators are per validator while the total amount which is going into the checkpoint computation is across the whole host zone.

Reply:
Can you elaborate on how a malicious attacker could game this? Keep in mind, a slash would have had to occur here for the attack to be relevant

Reply:
and even if it was manipulated such that the old progress was decreased, a LS of more than 1% of TVL would still trip the query

Reply:
Specifically I disagree with statement 2, I think they can cause this with an arbitrarily small unbonding, it doesn't have to be greater than the amount which is being new staked.

So the reason we need to do the ICQ at all is a validator could have been slashed and we need to know the new exchange rate in order to give out the right amount of stTokens to new deposits. The nature of the attack is that attacker notices (or causes) a validator to be slashed before the protocol has yet noticed and then by keeping the amounts correct they can deposit arbi.  For simplicity imagine not many other people are also depositing or unbonding from this target validator. So even if unbonding ICAs only happen every 3-5 days, they know when the next 3-5 day unbonding ICA is going to happen and by waiting until shortly before it they also know how much.  They can then use LSM tokens to add as much value to the specific slashed validator so that it is right before triggering the ICQ, when the unbonding ICAs happen the total amount delegated across the whole hostzone will decrease.  Even a small decrease in the total will lead to the "checkpoint" going down and the old/checkpoint and new/checkpoint now falling onto above the line on the next deposit.  They can then do this same thing again and again to continue to deposit onto this validator before we have checked the ICQ to see if it is slashed.  

You are correct that other people can come in and mess up the attackers plans like if another individual does more than 1% in a single transaction to this specific validator.